src/network/gameclient.cc

```diff
@@ -1150,17 +1158,19 @@ void GameClient::handle_network() {
 	}
 	try {
 		assert(d->net != nullptr);
-		// Check if the connection is still open
-		if (!d->net->is_connected()) {
-			disconnect("CONNECTION_LOST", "", false);
-			return;
-		}
 		// Process all available packets
 		std::unique_ptr<RecvPacket> packet = d->net->try_receive();
 		while (packet) {
 			handle_packet(*packet);
 			packet = d->net->try_receive();
 		}
+		// Check if the connection is still open
+		if (!d->net->is_connected()) {
+			disconnect("CONNECTION_LOST", "", false);
+			return;
+		}
+	} catch (const WLWarning&) {
+		throw;
```

Comment:
Is it clear that the client is disconnected upon a WLWarning? Since you have a guard in `disconnect()` now you may call it here again to be sure.